### PR TITLE
Timeout

### PR DESCRIPTION
**Pull request by macetheace96**
fixed bug where compute returned empty dataframe when called with timeout. There were issues with exception handling that came up, but these have hopefully been resolved as well. Timeout still only returns metafeatures that were computed within the time limit, no placeholder values are added.

### CODE REVIEW

**Comment by bjschoenfeld** [comment]
Please merge develop into timeout branch to resolve conflicts.

**Comment by bjschoenfeld** [comment]
This code does not run. Please make sure all tests pass before ever pushing code (assuming tests exist, haha). 

There are conflicts with the target branch. Please merge develop into your branch and resolve the conflicts.

When these are fixed, I will review this again.

**Comment by macetheace96** [comment]
multiprocessing commits have been moved to branch "multiprocessing" and timeout has been reverted to commit "stable timeout function"

File: test/metalearn/metafeatures/test_metafeatures.py [hunk 1]
```diff
@@ -298,7 +298,7 @@ def metafeatures_suite():
     test_cases = [MetaFeaturesTestCase, MetaFeaturesWithDataTestCase]
     return unittest.TestSuite(map(unittest.TestLoader().loadTestsFromTestCase, test_cases))
     # suite = unittest.TestSuite()
-    # suite.addTest(MetaFeaturesTestCase("test_dataframe_input_error"))
+    # suite.addTest(MetaFeaturesWithDataTestCase("test_timeout"))
     # return suite
```
Comment (bjschoenfeld) [inline test/metalearn/metafeatures/test_metafeatures.py]: We should cleanup these comments.

File: metalearn/metafeatures/metafeatures.py [hunk 2]
```diff
@@ -86,57 +84,81 @@ def compute(
...
+    def _threadsafe_timeout_function(self, f, args, timeout, metafeature_ids):
+        p = multiprocessing.Process(target=f, args=args)
+        p.start()
+        try:
+            p.join(timeout)
+            if p.is_alive():
+                p.terminate()
+                p.join()
+        except multiprocessing.TimeoutError:
+            print("timeout")
```
Comment (bjschoenfeld) [inline metalearn/metafeatures/metafeatures.py]: we should not print anything on timeout, just return the TIMEOUT flag for any metafeatures not computed

**Pull request merged**